fastlane/spec/actions_specs/create_keychain_spec.rb

```diff
@@ -2,106 +2,122 @@
   describe Fastlane::FastFile do
     describe "Create keychain Integration" do
       it "works with name and password" do
+        keychain_password = "testpassword"
+        keychain_name = "test.keychain"
+        keychain_path = "~/Library/Keychains/#{keychain_name}"
         result = Fastlane::FastFile.new.parse("lane :test do
           create_keychain ({
-            name: 'test.keychain',
-            password: 'testpassword',
+            name: '#{keychain_name}',
+            password: '#{keychain_password}',
           })
         end").runner.execute(:test)
 
         expect(result.size).to eq(3)
-        expect(result[0]).to eq('security create-keychain -p testpassword ~/Library/Keychains/test.keychain')
-
-        expect(result[1]).to start_with('security set-keychain-settings')
-        expect(result[1]).to include('-t 300')
-        expect(result[1]).to_not(include('-l'))
-        expect(result[1]).to_not(include('-u'))
-        expect(result[1]).to include('~/Library/Keychains/test.keychain')
+        expect(result[0]).to eq("security create-keychain -p #{keychain_password.shellescape} #{keychain_path.shellescape}")
+        expect(result[1]).to start_with("security set-keychain-settings")
+        expect(result[1]).to include("-t 300")
+        expect(result[1]).to_not(include("-l"))
+        expect(result[1]).to_not(include("-u"))
+        expect(result[1]).to include(keychain_path.to_s)
         expect(result[2]).to start_with("security list-keychains -s")
-        expect(result[2]).to end_with(File.expand_path('~/Library/Keychains/test.keychain').to_s)
+        expect(result[2]).to end_with(File.expand_path(keychain_path.to_s).to_s)
```

Comment:
@janpio Note that I did not change this to `keychain_path.shellescape` because of the link in the link causes the keychain path to be expanded:
https://github.com/fastlane/fastlane/pull/14311/files#diff-0231a345cb1ba3f328959032b555024cR57

So `~/Library/Keychains/...` becomes `/Users/foo/Library/Keychains/...` in this test case. It seems a little odd so if you have any thoughts, let me know.

Reply:
@janpio Nevermind the note above, I realized my mistake and resolved the use of `.shellescape` in this test case.